### PR TITLE
Fix revert to live 30

### LINKED ISSUE #4711: "Revert to live" not working for brand new pages
If you create a new page and then click "History" -> "Revert to live" you'll get HTTP 500.

The problem can be reproduced using the online demo at http://www.django-cms.org

### PR DESCRIPTION
**Pull request by mkoistinen**
Fixes #4711 and #4712.

This should be forward ported to 3.1.x and 3.2.x.

### CODE REVIEW

File: cms/admin/pageadmin.py [hunk 1]
```diff
@@ -1206,10 +1206,15 @@ def revert_page(self, request, page_id, language):
...
 
-        referer = request.META.get('HTTP_REFERER', '')
-        path = '../../'
-        if admin_reverse('index') not in referer:
-            path = '%s?%s' % (referer.split('?')[0], get_cms_setting('CMS_TOOLBAR_URL__EDIT_OFF'))
+        try:
+            path = page.get_absolute_url(language=language)
+            path = '%s?%s' % (
+                path, get_cms_setting('CMS_TOOLBAR_URL__EDIT_OFF'))
+        except:
```
Comment (czpython) [inline cms/admin/pageadmin.py]: I don't really like this. Did the ```get_absolute_url``` fail at any point?

Comment (mkoistinen) [inline cms/admin/pageadmin.py]: Yes, in my testing, it failed during the page-creation process.

Comment (yakky) [inline cms/admin/pageadmin.py]: I cannot reproduce the ``get_absolute_url`` failure. How can I trigger that?

**Comment by yakky** [comment]
@mkoistinen this should run fine. But let me do other tests and let me merge this when I feel confident we're not breaking things

**Pull request merged**